discovery/gossiper_test.go

```diff
@@ -202,6 +202,27 @@ func (r *mockGraphSource) ForEachNode(func(node *channeldb.LightningNode) error)
 
 func (r *mockGraphSource) ForAllOutgoingChannels(cb func(i *channeldb.ChannelEdgeInfo,
 	c *channeldb.ChannelEdgePolicy) error) error {
+
+	chans := make(map[uint64]channeldb.ChannelEdge)
+	for _, info := range r.infos {
+		info := info
+
+		edgeInfo := chans[info.ChannelID]
+		edgeInfo.Info = &info
+		chans[info.ChannelID] = edgeInfo
+	}
+	for _, edges := range r.edges {
+		edges := edges
+
+		edge := chans[edges[0].ChannelID]
```

Comment:
is the source always the smaller of the two pubkeys?

Reply:
Hmm, the way we have things set up atm, we always just assume that the gossiper itself is the "first" key. I think this would only be an issue if we had an actual router sub-system hooked up, as that's what would commit this ordering to the index. FWIW, all the existing tests seem to be built around this assumption, and they all seem to run relatively reliably. I think this is in part due to the deterministic way we always assign ourselves to the first slot, and the remote party to the second slot.

Reply:
sounds good, the keys are generated randomly so we should know pretty quickly if this assumption is ever invalidated